src/main/java/uk/ac/ebi/eva/pipeline/io/mappers/VariantVcfFactory.java

```diff
@@ -140,99 +120,68 @@
-        } else {
-            int start = position + indexOfDifference;
-            int end = position + reference.length() - 1;
-            String ref = reference.substring(indexOfDifference);
-            String inAlt = alt.substring(indexOfDifference);
-            return new VariantKeyFields(start, end, ref, inAlt);
+    private String removeChrPrefix(String chromosome) {
+        boolean ignoreCase = true;
+        int startOffset = 0;
+        String prefixToRemove = "chr";
+        if (chromosome.regionMatches(ignoreCase, startOffset, prefixToRemove, startOffset, prefixToRemove.length())) {
+            return chromosome.substring(prefixToRemove.length());
         }
+        return chromosome;
     }
 
-    protected VariantKeyFields createVariantsFromInsertionEmptyRef(int position, String alt) {
-        return new VariantKeyFields(position, position + alt.length() - 1, "", alt);
-    }
+    private List<VariantKeyFields> buildVariantKeyFields(int position, String reference, String[] alternateAlleles) {
+        List<VariantKeyFields> generatedKeyFields = new ArrayList<>();
+
+        for (int i = 0; i < alternateAlleles.length; i++) { // This index is necessary for getting the samples where the mutated allele is present
+            VariantKeyFields keyFields = normalizeSuffixFirst(position, reference, alternateAlleles[i]);
+            keyFields.setNumAllele(i);
 
-    protected VariantKeyFields createVariantsFromDeletionEmptyAlt(int position, String reference) {
-        return new VariantKeyFields(position, position + reference.length() - 1, reference, "");
+            // Since the reference and alternate alleles won't necessarily match
+            // the ones read from the VCF file but they are still needed for
+            // instantiating the variants, they must be updated
+            alternateAlleles[i] = keyFields.alternate;
+            generatedKeyFields.add(keyFields);
+        }
+        return generatedKeyFields;
     }
 
     /**
-     * Calculates the start, end, reference and alternate of an indel where the
-     * reference and the alternate are not empty.
+     * Calculates the start, end, reference and alternate of a variant where the
+     * reference and the alternate are not identical.
      * <p>
      * This task comprises 2 steps: removing the trailing bases that are
      * identical in both alleles, then the leading identical bases.
      *
      * @param position Input starting position
      * @param reference Input reference allele
-     * @param alt Input alternate allele
+     * @param alternate Input alternate allele
      * @return The new start, end, reference and alternate alleles
      */
-    protected VariantKeyFields createVariantsFromIndelNoEmptyRefAlt(int position, String reference, String alt) {
-        int indexOfDifference;
+    protected VariantKeyFields normalizeSuffixFirst(int position, String reference, String alternate)
+            throws NotAVariantException {
```

Comment:
I think we have a skip listener, and we allow up to 50 variants to fail. without the listener, the first no-variant would make the step fail.